### DISCUSSION TITLE
Updating the client cache

### DISCUSSION THREAD

**Discussion by lensbart** [remix-run/remix#4582] [Q&A]
I’m migrating an app from using Apollo Client to Remix. As mentioned in other discussions, the idiomatic way to do so is by using a lightweight GraphQL client (in my case `graphql-request`) in the loaders and getting rid of Apollo Client entirely.

This approach works well, for the most part. Where it doesn’t shine is in making small updates to pages that contain a large amount of data. With Apollo, it would be possible to imperatively update the cache (just the part of it that gets updated), and not refetch the entire page contents. With Remix, it seems impossible to manually update the client cache. Instead, the entire page must be refetched after every mutation.

* For now, the best approach seems to be using a server-side cache, so that expensive queries won’t have to rerun, but this feels more like a band-aid solution and would still be wasteful in terms of bytes sent.
* `unstable_shouldReload` only allows to prevent refetching at the loader level, but doesn’t help if a small part a large loader payload is being modified by the mutation.

It seems that it would be “philosophically in line” with this project if we could just fetch the minimum necessary amount of data (i.e. just use the action/mutation response to update the client, not rerun any loader).

Is this something that is being considered for future inclusion? Or am I missing something and is this already possible? Or should I take a different approach altogether?

Thank you!

**Comment by lensbart** [comment]
After giving it some thought, the most favourable approach seems to be to instantiate `ApolloClient` on the client, but only for the subtree of the app where the atomic edits to the client cache are desired, which (thankfully) is just one route. Only this route is wrapped in `ApolloProvider`.

**Comment by reillyjodonnell** [comment]
This is an understated problem imo
